### ISSUE TITLE
Tekla: Speckle causing System.AccessViolationException on closing Tekla

### ISSUE THREAD

**Issue by eestrado** [specklesystems/speckle-sharp#1203]
On closing Tekla after running Speckle, Tekla crashes with the following error:

```
System.AccessViolationException
  HResult=0x80004003
  Message=Attempted to read or write protected memory. This is often an indication that other memory is corrupt.
  Source=<Cannot evaluate the exception source>
  StackTrace:
<Cannot evaluate the exception stack trace>

  This exception was originally thrown at this call stack:
    Tekla.Structures.ComponentCatalog.CatalogObjects.CatalogPluginItem.Invoke()
    CatalogTree.CatalogTreeViewModel.InvokeCommand(object, bool)
    CatalogTree.CatalogTreePanel.CheckForAndHandleHitOnItemPlaceholder(Tekla.Structures.ComponentCatalog.CatalogInterfaces.ICatalogPlaceholder)
    CatalogTree.CatalogTreePanel.HandleLeftButtonSingleClick(Tekla.Structures.ComponentCatalog.CatalogInterfaces.ICatalogPlaceholder, System.Windows.Point)
    CatalogTree.CatalogTreePanel.OnSingleClick(System.Windows.Input.MouseButtonEventArgs, Tekla.Structures.ComponentCatalog.CatalogInterfaces.ICatalogPlaceholder)
    CatalogTree.CatalogTreePanel.OnMouseUp(System.Windows.Input.MouseButtonEventArgs)
    System.Windows.UIElement.OnMouseUpThunk(object, System.Windows.Input.MouseButtonEventArgs)
    System.Windows.RoutedEventArgs.InvokeHandler(System.Delegate, object)
    System.Windows.RoutedEventHandlerInfo.InvokeHandler(object, System.Windows.RoutedEventArgs)
    System.Windows.EventRoute.InvokeHandlersImpl(object, System.Windows.RoutedEventArgs, bool)
    ...
    [Call Stack Truncated]
```

To reproduce: 
- Open Tekla
- Attach debugger to process
- Run Speckle
- Close Tekla

I think it might have to do with the override of the OnClosing event in the DesktopUI2.Views.MainWindow class:
```
protected override void OnClosing(CancelEventArgs e)
    {
      this.Hide();
      e.Cancel = true;
      base.OnClosing(e);
    }
```
```e.Cancel = true``` causes the window to never actually close even though Tekla tries to shut it down on close and might still be trying to access some resources in Tekla even after they are disposed.

**Comment by eestrado** [comment]
Looks to be related to issue #1192 maybe same resolution can be applied to Tekla.

**Comment by teocomi** [comment]
Indeed, we were not correctly initializing Avalonia, this should do it: https://github.com/specklesystems/speckle-sharp/commit/2527da1ad4efe331ccefcd4217aab2a62a79bf7a

**Comment by AlanRynne** [comment]
Seems like we forgot to close this, since it was merged in #1204.

Anyway, a more general solution will come when Avalonia's fix gets released. Which is tracked in #1192
